common/lib/client/realtimepresence.js

```diff
@@ -9,6 +9,15 @@ var RealtimePresence = (function() {
 		return realtimePresence.channel.realtime.auth.clientId;
 	}
 
+	function isAnonymous(realtimePresence) {
+		var realtime = realtimePresence.channel.realtime;
+		/* If not currently connected, we can't assume that we're an anonymous
+		 * client, as realtime may inform us of our clientId in the CONNECTED
+		 * message. So assume we're not anonymous and leave it to realtime to
+		 * return an error if we are */
+		return !realtime.auth.clientId && realtime.connection.state === 'connected';
```

Comment:
This does work, but I am not sure I agree entirely with this logic.  If for example, a user has provided a `clientId` in the `ClientOptions`, but not yet connected, then we can assume the `clientId` is specified anyway because if the `CONNECTED` message comes back with a different `clientId`, the connection fails at that point anyway.  I think this will do, but it's not 100% right.